commerce-service/src/main/java/com/liferay/commerce/internal/upgrade/v4_6_0/CommerceOrderAddressUpgradeProcess.java

```diff
@@ -0,0 +1,298 @@
+
+			// Create columns on CommerceOrder for placed order address info
+
+			_addOrderAddressColumns();
+
+			// Add to new billing address columns for placed orders
+
+			String updateOrderBillingAddressSQL = StringBundler.concat(
+				"update CommerceOrder set billingName = ?, billingDescription ",
+				"= ?, billingStreet1 = ?, billingStreet2 = ?, billingStreet3 ",
+				"= ?, billingCity = ?, billingZip = ?, billingRegionId = ?, ",
+				"billingCountryId = ?, billingPhoneNumber = ?, ",
+				"billingAddressId = 0 where billingAddressId = ?");
+
+			_updateAddressValues(rs, updateOrderBillingAddressSQL);
+
+			// Add to new Shipping address columns for placed orders
+
+			String updateOrderShippingAddressSQL = StringBundler.concat(
+				"update CommerceOrder set shippingName = ?, ",
+				"shippingDescription = ?, shippingStreet1 = ?, ",
+				"shippingStreet2 = ?, shippingStreet3 = ?, shippingCity = ?, ",
+				"shippingZip = ?, shippingRegionId = ?, shippingCountryId = ?",
+				", shippingPhoneNumber = ?, shippingAddressId = 0 where ",
+				"shippingAddressId = ?");
+
+			_updateAddressValues(rs, updateOrderShippingAddressSQL);
+
+			// Create columns on Shipment to hold shipping address info
+
+			_addShipmentAddressColumns();
+
+			// Add to new Shipping address columns for shipments
+
+			String updateShipmentAddressSQL = StringBundler.concat(
+				"update CommerceShipment set shippingName = ?,",
+				"shippingDescription = ?, shippingStreet1 = ?, ",
+				"shippingStreet2 = ?, shippingStreet3 = ?, shippingCity = ?, ",
+				"shippingZip = ?, shippingRegionId = ?, shippingCountryId = ?,",
+				"shippingPhoneNumber = ? where commerceAddressId = ?");
+
+			_updateAddressValues(rs, updateShipmentAddressSQL);
+
+			// Remove all addresses that belong to orders
+
+			try (Statement s1 = connection.createStatement(
+					ResultSet.TYPE_FORWARD_ONLY, ResultSet.CONCUR_UPDATABLE)) {
+
+				s1.executeUpdate(
+					"delete from CommerceAddress where classNameId = " +
+						commerceOrderClassNameId);
+			}
+
+			// Drop stale column from CommerceShipment
+
+			_dropColumn(
+				CommerceShipmentModelImpl.TABLE_NAME, "commerceAddressId");
+		}
+	}
+
+	private void _addColumn(
+			Class<?> entityClass, String tableName, String columnName,
+			String columnType)
+		throws Exception {
+
+		if (_log.isInfoEnabled()) {
+			_log.info(
+				String.format(
+					"Adding column %s to table %s", columnName, tableName));
+		}
+
+		if (!hasColumn(tableName, columnName)) {
+			alter(
+				entityClass,
+				new AlterTableAddColumn(
+					columnName + StringPool.SPACE + columnType));
+		}
+		else {
+			if (_log.isInfoEnabled()) {
+				_log.info(
+					String.format(
+						"Column %s already exists on table %s", columnName,
+						tableName));
+			}
+		}
+	}
+
+	private void _addOrderAddressColumns() throws Exception {
+		_addColumn(
+			CommerceOrderModelImpl.class, CommerceOrderModelImpl.TABLE_NAME,
+			"billingName", "STRING");
+		_addColumn(
```

Comment:
watch the field types here -- this should match what's in tables.sql. Looks like at least many of these `STRING`s should be `VARCHAR(75)`s.